src/pubsub/mod_pubsub.erl

```diff
@@ -1313,18 +1312,16 @@ iq_pubsub(Host, ServerHost, From, IQType, #xmlel{children = SubEls} = QueryEl,
           Lang, Access, Plugins) ->
     case xml:remove_cdata(SubEls) of
         [#xmlel{name = Name} = ActionEl | _] ->
-            report_iq_action_metrics_before_result(ServerHost, IQType, Name),
             Node = exml_query:attr(ActionEl, <<"node">>, <<>>),
-            {Time, Result} = timer:tc(fun iq_pubsub_action/6,
-                                      [IQType, Name, Host, Node, From,
-                                       #{server_host => ServerHost,
-                                         plugins => Plugins,
-                                         access => Access,
-                                         action_el => ActionEl,
-                                         query_el => QueryEl,
-                                         lang => Lang}]),
-            report_iq_action_metrics_after_return(ServerHost, Result, Time, IQType, Name),
-            Result;
+            mongoose_instrument:span(event_name(IQType, Name), #{host_type => ServerHost}, fun iq_pubsub_action/6,
+                                     [IQType, Name, Host, Node, From,
+                                      #{server_host => ServerHost,
+                                        plugins => Plugins,
+                                        access => Access,
+                                        action_el => ActionEl,
+                                        query_el => QueryEl,
+                                        lang => Lang}],
```

Comment:
Maybe construct this map separately to simplify this big expression?

Reply:
I've moved this map in both places, please see if this is something that you've had in mind. I think it looks better now indeed.